templates/types/streaming/fastapi/app/api/routers/vercel_response.py

```diff
@@ -37,17 +29,47 @@ def convert_data(cls, data: dict):
         data_str = json.dumps(data)
         return f"{cls.DATA_PREFIX}[{data_str}]\n"
 
+    def __init__(
+        self,
+        request: Request,
+        event_handler: EventCallbackHandler,
+        response: StreamingAgentChatResponse,
+        chat_data: ChatData,
+    ):
+        content = VercelStreamResponse.content_generator(
+            request, event_handler, response, chat_data
+        )
+        super().__init__(content=content)
+
     @classmethod
     async def content_generator(
         cls,
         request: Request,
         event_handler: EventCallbackHandler,
         response: StreamingAgentChatResponse,
+        chat_data: ChatData,
     ):
         # Yield the text response
         async def _chat_response_generator():
+            final_response = ""
             async for token in response.async_response_gen():
-                yield cls.convert_text(token)
+                final_response = token
+                yield VercelStreamResponse.convert_text(token)
+
+            # Generate questions that user might interested to
+            questions = await NextQuestionSuggestion.suggest_next_questions(
+                chat_data.messages
```

Comment:
is there a big difference in the result if we add all messages or just the last two messages?

Reply:
Not a big difference for me, can reduce it to just last two messages or even just a message from assistant (with a minor adjustment in the prompt) then the generated questions are still good for me.